crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1219,3 +1214,40 @@ pub struct LockDetails {
     pub epoch: EpochId,
     pub tx_digest: TransactionDigest,
 }
+
+/// Identifies an input to a transaction that is missing in the store
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, Ord, PartialOrd)]
+pub enum MissingInput {
```

Comment:
My preference would be to use `Option<VersionNumber>` for version which would be `None` for move packages. It may be easier to deal with but up to you.